### DISCUSSION TITLE
Unable to Insert in to Postgres Table

### DISCUSSION THREAD

**Discussion by JJWest12** [nextgenhealthcare/connect#5205] [Q&A]
I am trying to work with writing to a postgres database for the first time however I am running in to some errors in my channel.

To start, I am only messing around with a test table that is designed as such:

Schema Name: experiments
Table Name: TESTING
Columns: TEST (text)

My channel is set up to read a file and do some QA checking which is all working as intended. I then have a database writer destination that is able to successfully connect to the db (I confirmed that the table I want to write to shows up when I click the Generate: Insert template button) and I try to run this simple insert statement. This insert statement has nothing to do with the file reading part for now, it will eventually but I want to get writing to it successfully working first.

INSERT INTO TESTING (TEST) VALUES ('DOES THIS WORK?')

I am getting this error on the destination:

Caused by: org.postgresql.util.PSQLException: ERROR: relation "testing" does not exist

Does anyone know what might be causing this error and why I can't get a simple insert to work for this table?

**Comment by pacmano1** [comment]
I would expect that is a case sensitivity problem with the table name.  If the table name is truly in all capitals you need to wrap it in double quotes.  

Or, as I usually say, when you build database schemas do everything in all lowercase to avoid this kind of thing.  

This problem will also affect your column names.

**Reply by joaryche** [reply]
I would expect he has not connected to the correct database, 'experiments'.   He might have just connected to the postgres default database where his TESTING table is not present.

**Reply by pacmano1** [reply]
![image](https://user-images.githubusercontent.com/44065187/169317413-5cce20d5-0739-40de-bec1-7c8c1677e575.png)

this does not work:

![image](https://user-images.githubusercontent.com/44065187/169317799-0548ea3a-76e7-4ba2-b520-5edb471586e7.png)

**Reply by pacmano1** [reply]
@JJWest12 What was the final solution?

**Reply by mjtrac** [reply]
I've seen this exact situation, where the "Generate" button gets me a list of tables and fields but I then get told the relation doesn't exist when sending a command.  I no longer have a history, but it was Postgres security.  I think I changed ownership of the db and/or schema and/or table to the same Postgres user Mirth was using in its credentials.

**Reply by pacmano1** [reply]
I still expect it is this issue I just submitted: https://github.com/nextgenhealthcare/connect/issues/5208

**Reply by mjtrac** [reply]
I'm aware that it "makes no sense" that the table retrieval works but then there is a rejection when SQL is sent.  But it happened to me only a day or so before I commented, and I resolved it on the Postgres side.  I will attempt to replicate, and I can understand why this would be doubted unless/until I can replicate it.

**Reply by mjtrac** [reply]
OK, here's what I'm seeing and you are correct.  I have a table Test1, it shows as Test1 in pgAdmin4, shows as Test1 in the "Generate" dialog, and the Generate dialog generates a select that refers to Test1.  The error message on executing the SQL is ERROR: relation "test1" does not exist.  I must have changed something else when I changed my SQL security (this is all in a playpen) and I was surprised when the change to SQL security seemed to have worked.  But among other things, I apparently created a table test2 (t not T), and perhaps that's what i used after the change.  Apologies for having introduced any confusion.

**Reply by JJWest12** [reply]
Sorry I am super late on this, it turns out my issue was that I needed to include the schema name in the database in order for it to work. So in my case, rather than:

INSERT INTO TESTING (TEST) VALUES ('DOES THIS WORK?')

I needed to do:

INSERT INTO experiments.TESTING (TEST) VALUES ('DOES THIS WORK?')

**Reply by pacmano1** [reply]
Glad you solved that - however I am still puzzled this worked wihtout double quotes unless your postgres DB is set to a non-typical collation.

**Reply by JJWest12** [reply]
The whole "test" thing was just an example as to not expose my actual data names and such, so the lack of double quotes was just me mistakenly free typing. I am however stuck on a new issue that is puzzling me now...

	var query = "SELECT a, b, c, d, e, f, g, h, i, j, k, l, m, n FROM schema.tablename WHERE z = 'Y'"
	result = dbConn.executeCachedQuery(query);
	
Getting a  ERROR: syntax error at or near "Y"?..

**Reply by pacmano1** [reply]
When you create queries, put them in $c, then in the message viewer cut and paste that vaue to your SQL editor, which usually makes the problem transparent.

```JS
var query = "SELECT a, b, c, d, e, f, g, h, i, j, k, l, m, n FROM schema.tablename WHERE z = 'Y'";
$c('SQLQuert',query);
var result = dbConn.executeCachedQuery(query);
```

**Reply by JJWest12** [reply]
That's the weird part. Did that and the query is working just fine in pgAdmin

**Reply by pacmano1** [reply]
maybe your dbConn is wrong.  I would also try stripping the `schema.`

**Reply by JJWest12** [reply]
I would agree with that however the issue being that when I remove the where clause everything works fine. It has to have something to do with that

**Reply by pacmano1** [reply]
Dunno, since you didn't paste a schema real or not.

And on that note your "lack of double quotes was just me mistakenly free typing." is a disservice to those trying to help you by causing them to dive down peraps the incorrect path.
